src/Meshes/mesh.ts

```diff
@@ -4508,6 +4508,90 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
+     * @param scene defines the hosting scene 
+     * @returns Geodesic mesh
+     */
+     public static CreateGeodesic(
+        name: string,
+        options: {
+            m?: number;
+            n?: number;
+            size?: number;
+            sizeX?: number;
+            sizeY?: number;
+            sizeZ?: number;
+            flat?: boolean;
+            faceUV?: Vector4[];
+            faceColors?: Color4[];
+            updatable?: boolean;
+            sideOrientation?: number;
+        },
+        scene: Scene
+    ): Mesh {
+        throw _DevTools.WarnImport("MeshBuilder");
+    }
+
+    /**
+     * Creates the Mesh for a Goldberg Polyhedron
+     * @param name defines the name of the mesh
+     * @param options an object used to set the following optional parameters for the polyhedron, required but can be empty
+     * * m number of horizontal steps along an isogrid
+     * * n number of angled steps along an isogrid
+     * * size the size of the Goldberg, optional default 1
+     * * sizeX allows stretching in the x direction, optional, default size
+     * * sizeY allows stretching in the y direction, optional, default size
+     * * sizeZ allows stretching in the z direction, optional, default size
+     * * faceUV an array of Vector4 elements used to set different images to the top, rings and bottom respectively
+     * * faceColors an array of Color3 elements used to set different colors to the top, rings and bottom respectively
+     * * subdivisions increasing the subdivisions increases the number of faces, optional, default 4
+     * * sideOrientation optional and takes the values : Mesh.FRONTSIDE (default), Mesh.BACKSIDE or Mesh.DOUBLESIDE
+     * * frontUvs only usable when you create a double-sided mesh, used to choose what parts of the texture image to crop and apply on the front side, optional, default vector4 (0, 0, 1, 1)
+     * * backUVs only usable when you create a double-sided mesh, used to choose what parts of the texture image to crop and apply on the back side, optional, default vector4 (0, 0, 1, 1)
+     * @param scene defines the hosting scene 
+     * @returns Goldberg mesh
+     */
+     public static CreateGoldberg(
+        name: string,
+        options: {
+            m?: number;
+            n?: number;
+            size?: number;
+            sizeX?: number;
+            sizeY?: number;
+            sizeZ?: number;
+            updatable?: boolean;
+            sideOrientation?: number;
+        },
+        scene: Scene
+    ): Mesh {
+        throw _DevTools.WarnImport("MeshBuilder");
+    }
+
+    /**
+     * Extends a mesh to a Goldberg mesh
+     * @param mesh the mesh to convert
+     * Warning  the mesh to convert MUST be an import of a peviously exported Goldberg mesh     
+     */
+    public static ToGoldberg(mesh: Mesh) {
+    }

```

Comment:
Should probably have a warning here: throw _DevTools.WarnImport("MeshBuilder");

Reply:
MeshBuilder is not an alternative here. This takes a mesh that is an import of a previously exported Goldberg mixin extension to a mesh and adds back the mixin. As such I believe that 

BABYLON.Mesh.ToGoldberg(mesh)

makes more sense to the user than

BABYLON.MeshBuilder.ToGoldberg(mesh)

Reply:
Yup agree but still the user will need to import goldbergBuilder in order to have the function available which you should warn about here :-)

throw _DevTools.WarnImport("goldbergBuilder");

Reply:
I would actually even keep it completely outside of mesh and do not add a static here. It is not needed. We only did it in other case for back compat which is a pain for ES6 so far :-) @RaananW can confirm.

Reply:
> @RaananW sorry copy and paste error
> 
> https://github.com/BabylonJSGuide/Babylon.js/blob/master/src/Meshes/Builders/goldbergBuilder.ts
> 
> Lines 150 to 328 for mixin
> 
> From 330 for ToGoldberg
> 
> Line 113 uses mixin for CreateGoldberg

Look good to me! It feels like it can be a part of the mesh prototype (as opposed to a static), but you can leave it this way. as long as it is separated and can be tree-shaken it is totally fine.